Add AIP pages to dashboard

jraddaoui opened the pull request:
- Add browse and index AIP pages.
- Normalize SIP and Location pages:
  - Remove ID/# columns from tables.
  - Remove empty/TODO fields from tables and lists.
  - Remove location column from SIPs table.
  - Use AIP/SIP/Location details as index page headings.
- Consider AIP statuses in status badge component.
- Add `aip` store.
- Rename `storage` store to `location`.
- Rename `ingest` store to `sip`.
- Add tests for fetch current and fetch list to stores.
- Add hot module replacement update to all stores.
- Move monitor event handler outside of `sip` store.

Refs #1117.

djjuhasz commented on dashboard/src/components/AipLocationCard.vue:
```diff
@@ -5,32 +5,30 @@ import { storageServiceDownloadURL } from "@/client";
-const ingestStore = useIngestStore();
+const sipStore = useSipStore();
 
 let failed = ref<boolean | null>(null);
```
Huh, I wonder why `const failed = ref(false);` wasn't used here?  I don't see any need for failed to be `null` and it's common practice to make refs a constant.  I know this isn't your code, but it would be nice to simplify this.

djjuhasz commented on dashboard/src/pages/storage/locations/[id]/index.vue:
```diff
@@ -1,35 +1,28 @@
+import { useLocationStore } from "@/stores/location";
 
-const storageStore = useStorageStore();
+const storageStore = useLocationStore();
```
I think this variable name should be "locationStore"?

djjuhasz commented on dashboard/src/pages/storage/locations/index.vue:
```diff
@@ -5,14 +5,14 @@ import PageLoadingAlert from "@/components/PageLoadingAlert.vue";
 layoutStore.updateBreadcrumb([{ text: "Storage" }, { text: "Locations" }]);
 
-const storageStore = useStorageStore();
+const storageStore = useLocationStore();
```
Should be "locationStore"?

djjuhasz commented on dashboard/src/stores/auth.ts:
```diff
@@ -310,3 +310,7 @@ export const useAuthStore = defineStore("auth", {
   },
 });
+
+if (import.meta.hot) {
```
Ooh, Hot meta imports. :fire: :stuck_out_tongue:

djjuhasz reviewed (approved):
This looks great @jraddaoui! I had a couple of minor notes, but no blockers. :+1:

jraddaoui committed:
Add AIP pages to dashboard

- Add browse and index AIP pages.
- Normalize SIP and Location pages:
  - Remove ID/# columns from tables.
  - Remove empty/TODO fields from tables and lists.
  - Remove location column from SIPs table.
  - Use AIP/SIP/Location details as index page headings.
- Consider AIP statuses in status badge component.
- Add `aip` store.
- Rename `storage` store to `location`.
- Rename `ingest` store to `sip`.
- Add tests for fetch current and fetch list to stores.
- Add hot module replacement update to all stores.
- Move monitor event handler outside of `sip` store.

jraddaoui committed:
Move AIP location card to AIP page

- Place move/download operation checks and triggers in AIP store.
- Update AIP location card component to use AIP store.
- Render AIP location card in AIP page instead of in SIP page.
- Hide choose storage location button for now.
- Add related AIP card to SIP page, showing AIP UUID and view button.
